src/course-unit/context/iFrameContext.tsx

```diff
@@ -16,11 +16,12 @@ export const IframeProvider: React.FC = ({ children }: { children: ReactNode })
     iframeRef.current = ref.current;
   }, []);
 
-  const sendMessageToIframe = useCallback((messageType: string, payload: any) => {
+  const sendMessageToIframe = useCallback((messageType: string, payload: any, consumerWindow?: any) => {
```

Comment:
[curious]: Since we have a full understanding of what's in the payload, should we type the payload structure (`type: string`, `payload: {}`)?

```
    window.parent.postMessage(
        {
            type: '<TYPE_NAME>',
            payload: { ... }
        }, document.referrer
    );
```

Reply:
We don't pass `description` to messages to iframe, so I think it's not necessary to add it here.
We pass this value from legacy to MFE but not the other way around.

Reply:
Resolved